docs/_docs/persistence/change-data-capture.adoc

```diff
@@ -0,0 +1,129 @@
+|===
+
+== API
+
+=== `org.apache.ignite.cdc.CdcEvent`
+Single change of data reflected by `CdcEvent`.
+Let's get a closer look to the information provided:
+
+[cols="20%,80%",opts="header"]
+|===
+|Name |Description
+| `key()` | Key for the changed entry.
+| `value()` | Value for the changed entry. This method will return `null` if event reflects removal.
+| `cacheId()` | ID of the cache where change happens. Value equal to the `CACHE_ID` from `SYS.CACHES` link:../monitoring-metrics/system-views.adoc#_CACHES[system view]
+| `partition()` | Partition of the changed entry.
+| `primary()` | Flag to distinguish if operation happen on primary or backup node.
+| `version()` | `Comparable` version of the changed entry. Internally Ignite maintain ordered versions of each entry so any changes of the same entry can be sorted.
+|===
+
+=== `org.apache.ignite.cdc.CdcConsumer`
+
+Consumer of change events. Should be implemented by the user.
+[cols="20%,80%",opts="header"]
+|===
+|Name |Description
+| `void start(MetricRegistry)` | Invoked one time on the start of CDC application. `MetricRegistry` should be used to export consumer specific metrics.
+| `boolean onEvents(Iterator<CdcEvent> events)` | Main method that processes changes. When this method returns `true` state will be saved on the disk. State points to the event next to the last read. In case of any failure consumption will continue from the last saved state.
+| `void stop()` | Invoked one time on the stop of CDC application.
+|===
+
+== Metrics
+
+`ignite-cdc.sh` uses the same SPI to export metrics as Ignite.
+The following metrics provided by the application itself (additional metrics may be provided by the consumer):
+|===
+|Name |Description
+| CurrentSegmentIndex | Index of currently processed WAL segment.
+| CommittedSegmentIndex | Index of WAL segment contains last commited state.
+| CommittedSegmentOffset | Commited offset in bytes inside WAL segment.
+| LastSegmentConsumptionTime | Time in milliseconds when last segment processing was started.
+| BinaryMetaDir | Binary meta directory this application reads from.
+| MarshallerDir | Marshaller directory this application reads from.
+| CdcDir | CDC directory this application reads from.
+|===
+
+== Logging
+
+`ignite-cdc.sh` use the same logging configuration as Ignite node. The only difference is log written in file "ignite-cdc.log"
+
+== Lifecycle
+
+IMPORTANT: `ignite-cdc.sh` implements fail-fast approach. It will just fail in case of any error. Restart should be configured with the OS tools.
+
+ 1. Finds required shared directories. Values from provided `IgniteConfiguration` taken.
+ 2. Locks CDC directory.
+ 3. Loads saved state.
+ 4. Starts the consumer.
+ 5. Infinitely waits for new available segment and processes it.
+ 6. Stop the consumer in case of failure or receive stop signal.
```

Comment:
```suggestion
 1. Find the required shared directories. Take the values from the provided `IgniteConfiguration`.
 2. Lock the Change Data Capture directory.
 3. Load the saved state.
 4. Start the consumer.
 5. Infinitely wait for the newly available segment and process it.
 6. Stop the consumer in case of a failure or a received stop signal.
```